File: solutions/code-carl/ch04_string/T3_LC0151_ReverseWords.java
/**
 * Practice template for:
 * - LeetCode 151. Reverse Words in a String
 *   https://leetcode.cn/problems/reverse-words-in-a-string/
 */
import java.util.Arrays;
import java.util.Collections;
import java.util.List;
import java.util.StringJoiner;
import java.util.stream.Collectors;

public class T3_LC0151_ReverseWords {
    public String reverseWords(String s) {
        String[] split = s.trim().split(" ");
        List<String> collect = Arrays.stream(split)
                .filter(sp -> !sp.isEmpty()).
                collect(Collectors.toList());
        Collections.reverse(collect);
        return String.join(" ", collect);
    }
}

File: solutions/python/code-carl/ch04_string/t3_lc0151_reverse_words.py
"""c
https://leetcode.cn/problems/reverse-words-in-a-string/
"""


class Solution:
    def reverseWords(self, s: str) -> str:
        return " ".join(s.strip().split()[::-1])


if __name__ == "__main__":
    solution = Solution()
    assert solution.reverseWords("the sky is blue") == "blue is sky the"
    assert solution.reverseWords("  hello world  ") == "world hello"
    print("All tests passed.")
